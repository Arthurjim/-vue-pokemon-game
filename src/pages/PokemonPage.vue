<template>
    <h1 v-if="!pokemon">Buscando pókemon...</h1>
    <div v-else>
        <h2 class="title">¿Quién es ese pókemon?</h2>
        <PokemonPicture :show-pokemon="showPokemon" :pokemon-id="pokemon.id" />
        <PokemonOptions
            :pokemons="pokemonsArr"
            :pokemon-id="pokemon.id"
            :pokemon-selected="pokemonSelected"
            @selection="checkAnswer"
        />

        <template v-if="showAnswer">
            <h2 class="message">{{ message }}</h2>
            <button class="btn_again" @click="newGame">Nuevo Juego</button>
        </template>
    </div>
</template>

<script>
import PokemonOptions from "@/components/PokemonOptions.vue";
import PokemonPicture from "@/components/PokemonPicture.vue";
import getPokemonOptions from "@/helpers/getPokemonOptions.js";

export default {
    name: "PokemonPage",
    data() {
        return {
            pokemonsArr: [],
            pokemon: null,
            showPokemon: false,
            showAnswer: false,
            message: "",
            pokemonSelected:null
        };
    },
    components: {
        PokemonOptions,
        PokemonPicture,
    },
    methods: {
        async mixmethod() {
            this.pokemonsArr = await getPokemonOptions();
            const rndInt = Math.floor(Math.random() * 4);
            this.pokemon = this.pokemonsArr[rndInt];
        },
        checkAnswer(id) {
            // console.log(numero)
            
            if (this.pokemon.id == id) {
                this.showPokemon = true;
                this.message = `Correcto, ${this.pokemon.name}`;
                this.showAnswer = true;

            } else {
                this.message = `Oops, era ${this.pokemon.name}`;
                this.showPokemon = true;
                this.showAnswer = true;
            }
            this.pokemonSelected=id;

            console.log('hola')
        },
        newGame() {
            this.showAnswer = false;
            this.showPokemon = false;
            this.pokemon = null;
            this.pokemonsArr = [];
            this.correctPokemon = [];
            this.mixmethod();
            this.pokemonSelected=null;

        },
    },
    mounted() {
        this.mixmethod();
    },
    // onMo:{
    //     '$poks':{
    //       async handler(){
    //         this.pokemons = await getPokemonOptions();
    //         console.log(this.pokemons)
    //       },
    //       immediate: true
    //     }
    // }
};
</script>

<style scoped>
    template{
        background: blue;
    }
    .title {
        color: rgb(255, 238, 0);
        text-shadow: 0px 2px 5px rgb(3, 3, 216);
        font-size: 40px;
        font-family: "Supermercado One", cursive;
        letter-spacing: 2px;
    }
.message{
    background: rgba(255, 238, 0, 0.527);;
    border-radius: 5px;
    color: rgb(15, 15, 15);
    font-family: "Supermercado One", cursive;
    font-size: 30px;
    margin: auto;
    padding: 25px;
    text-align: center;
    width: 300px;

}
.btn_again{
    padding: 10px 20px;
    border-radius: 5px;
    color: white;
    background: red;
    margin-top: 10px;
    font-family: "Supermercado One", cursive;
    border: none;
}
.btn_again:hover{
    background: #ff1919c4;

}
</style>
