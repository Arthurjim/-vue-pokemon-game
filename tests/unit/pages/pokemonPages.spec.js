import { mount, shallowMount } from "@vue/test-utils";
import PokemonPage from '@/pages/PokemonPage'


import { pokemonsMock } from "../mocks/pokemons.mock";
describe("Pokemon pages", () => {
    let wrapper;
    // let mixPokemonArray;
    beforeEach(() => {
        // mixPokemonArray=  jest.spyOn(PokemonPage.methods, 'mixmethod')

        wrapper = shallowMount(<PokemonPage />)
    });

    test("Debe de hacer match con el snaptchott", () => {
        expect(wrapper.html()).toMatchSnapshot()
    });
    test('Debe de llamar mixPokemonArray al montar', () => { 
        const mixPokemonArray= jest.spyOn(PokemonPage.methods, 'mixmethod')
        shallowMount(PokemonPage) 
        expect(mixPokemonArray).toHaveBeenCalled()
     })

     test('debe de hacer match con el snapshor cuando cargan los elemenos', () => { 
         const wrapper = mount(PokemonPage,{
             data(){
                return {
                    pokemonsArr: pokemonsMock,
                    pokemon: pokemonsMock[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: "",
                    pokemonSelected:null
                };
             }
         })

         expect(wrapper.html()).toMatchSnapshot()
      })
      test('debe de mostrar los componentes de PokemonPicture y PokemonOptions', () => {

        const wrapper = shallowMount(PokemonPage,{
            data(){
               return {
                   pokemonsArr: pokemonsMock,
                   pokemon: pokemonsMock[0],
                   showPokemon: false,
                   showAnswer: false,
                   message: "",
                   pokemonSelected:null
               };
            }
        })
        expect(wrapper.html()).toMatchSnapshot()

        const pokemonPicture = wrapper.find('pokemon-picture-stub')
        const pokemonOptions =wrapper.find('pokemon-options-stub')
        
       expect(pokemonPicture.exists()).toBeTruthy()
       expect(pokemonOptions.exists()).toBeTruthy()

       expect(pokemonPicture.attributes('pokemonid')).toBe('1')
       expect(pokemonOptions.exists('pokemons')).toBe(true)


        // expect(wrapper.findComponent(PokemonPicture).exists()).toBe(true);
        // expect(wrapper.find('PokemonOptions').exists()).toBe(true);

        // console.log(wrapper.element())
        // PokemonPicture
        // PokemonPicture

        // PokemonPicture pokemonId = 5
      })
});
