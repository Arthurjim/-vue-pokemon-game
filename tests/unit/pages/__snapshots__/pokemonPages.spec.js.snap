// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon pages Debe de hacer match con el snaptchott 1`] = `<h1>Buscando pókemon...</h1>`;

exports[`Pokemon pages debe de hacer match con el snapshor cuando cargan los elemenos 1`] = `
<div>
  <h2 class="title">¿Quién es ese pókemon?</h2>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="pokemon"></div>
  <div class="options-container">
    <ul>
      <li><button class="">
          <!--v-if--> bulbasaur
          <!--v-if-->
        </button></li>
      <li><button class="">
          <!--v-if--> ivysaur
          <!--v-if-->
        </button></li>
      <li><button class="">
          <!--v-if--> venusaur
          <!--v-if-->
        </button></li>
      <li><button class="">
          <!--v-if--> charmander
          <!--v-if-->
        </button></li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;

exports[`Pokemon pages debe de mostrar los componentes de PokemonPicture y PokemonOptions 1`] = `
<div>
  <h2 class="title">¿Quién es ese pókemon?</h2>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]" pokemonid="1"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
